src/main.rs

```diff
@@ -37,10 +41,34 @@ use queue::Queue;
 const SNARE_CONF_SEARCH: &[&str] = &["/etc/snare.conf", "~/.snare.conf"];
 
 pub(crate) struct Snare {
+    /// The location of snare.conf; this file will be reloaded if SIGHUP is received.
+    conf_path: PathBuf,
+    /// The current configuration: note that this can change at any point due to SIGHUP.
     config: Mutex<Config>,
+    /// The current queue of incoming jobs.
     queue: Mutex<Queue>,
+    /// The read end of the pipe used by the httpserver and the SIGHUP handler to wake up the job
+    /// runner thread.
     event_read_fd: RawFd,
+    /// The write end of the pipe used by the httpserver and the SIGHUP handler to wake up the job
+    /// runner thread.
     event_write_fd: RawFd,
+    /// Has a SIGHUP event occurred? If so, the jobrunner will process it, and set this to false in
+    /// case future SIGHUP events are detected.
+    sighup_occurred: Arc<AtomicBool>,
+}
+
+impl Snare {
+    /// Check to see if we've received a SIGHUP since the last check. If so, we will reload the
+    /// config file. **Note that because snare has multiple threads, the config file can change at
+    /// any arbitrary point, not just after calling this function.**
```

Comment:
I don't understand the "not just after calling this function" part of the comment.

When else can the config change?

Reply:
There are two threads in snare, so the config can change in another thread even if it's not called `check_for_hup`.

Reply:
To be clear, the config may change when **neither** thread received a HUP?

Reply:
At the moment, the signal comes in, a global bool is set, and then `jobrunner` checks that bool and reloads the config if necessary. There are only two threads, so one of them has to handle it (and the config stuff is way too much for it to be signal safe, so the signal handler can't deal with it directly).

Reply:
OK understood. But the comment makes it sound like some other part of the program (not this function) is mutating the config. As I understand you mean to say instead that this function could be run in another thread.

Reply:
Is https://github.com/softdevteam/snare/pull/15/commits/3f03f446c341e2b9b852f4f2998203f4ba7022d6 better?

Reply:
Much clearer, thanks.